src/cudamatrix/cu-vector-speed-test.cc

```diff
@@ -73,6 +73,54 @@ template<typename Real> void TestCuVectorSum(int32 dim) {
             << dim << ", speed was " << gflops << " gigaflops.";
 }
 
+#if HAVE_CUDA == 1
+// This test choose the min length of vectors to be reduced on GPU.
+// Smaller vector will be copied to RAM and reduced on CPU.
+template<typename Real> void TestCuVectorSumChooseMinLength() {
+  BaseFloat time_in_secs = 0.02;
+  for (int dim = 100; dim < 1000000; dim = dim * 1.5 + 1 ) {
+    CuVector<Real> M(dim);
+    BaseFloat gflops, gflops_cpu;
+    Real result = 0, result_cpu = 0;
+    M.SetRandn();
+    {
+      Timer tim;
+      int32 iter = 0;
+      for (; tim.Elapsed() < time_in_secs; iter++) {
+        // Force GPU reduction
+        int dimBlock = CU1DBLOCK;
+        int dimGrid = n_blocks(M.Dim(), dimBlock);
+        if (dimGrid > 256) {
+          dimGrid = 256;
+        }
+        CuVector<Real> ans(dimGrid, kUndefined);
+        cuda_vec_sum(dimGrid, dimBlock, M.Data(), ans.Data(), M.Dim(), 1);
+        CU_SAFE_CALL(cudaGetLastError());
+        Vector<Real> ans_cpu(ans);
+        result = ans_cpu.Sum();
+      }
+
+      BaseFloat fdim = dim;
+      gflops = (fdim * iter) / (tim.Elapsed() * 1.0e+09);
+    }
+    {
+      Timer tim;
+      int32 iter = 0;
+      for (; tim.Elapsed() < time_in_secs; iter++) {
+        Vector<Real> M_cpu(M);
+        result_cpu = M_cpu.Sum();
+      }
+
+      BaseFloat fdim = dim;
+      gflops_cpu = (fdim * iter) / (tim.Elapsed() * 1.0e+09);
+    }
+    KALDI_LOG<< "CuVector::Sum" << NameOf<Real>() << ", dim: " << dim
```

Comment:
can you please make the whitespace here more normal, e.g. space before <<, and line things up?